kubernetes_state/datadog_checks/kubernetes_state/kubernetes_state.py

```diff
@@ -383,6 +383,8 @@ def kube_pod_container_status_waiting_reason(self, message, **kwargs):
                     tags.append(self._format_tag("kube_container_name", label.value))
                 elif label.name == "namespace":
                     tags.append(self._format_tag(label.name, label.value))
+                elif label.name == "pod":
```

Comment:
I guess `kube_pod_container_status_waiting_reason()` and `kube_pod_container_status_terminated_reason()` could use some refactoring to avoid so much duplicated code between them as only the metric name and whitelist differ from one to the other.